### ISSUE TITLE
Class library has lost all intellisense in VS2017 after auto-migration

### ISSUE THREAD

**Issue by svallis** [aspnet/Mvc#5975]
After migrating a VS2015 MVC Core application with two projects (web app and class library) to VS2017 I've lost intellisense on all views within the class library. Pretty much everything in every single view is broken, so I'm sure it's something fairly basic that the migration tool didn't take care of for me. Even the `@model` directive in each razor view is an error.

The `csproj` looks like this:

    <Project Sdk="Microsoft.NET.Sdk">
    
      <PropertyGroup>
        <TargetFramework>netcoreapp1.1</TargetFramework>
        <AssemblyName>Library</AssemblyName>
        <PackageId>Library</PackageId>
        <PackageTargetFallback>$(PackageTargetFallback);dnxcore50</PackageTargetFallback>
        <RuntimeFrameworkVersion>1.1.1</RuntimeFrameworkVersion>
        <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>
        <GenerateAssemblyDescriptionAttribute>false</GenerateAssemblyDescriptionAttribute>
        <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
        <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
        <GenerateAssemblyCopyrightAttribute>false</GenerateAssemblyCopyrightAttribute>
        <GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>
        <GenerateAssemblyFileVersionAttribute>false</GenerateAssemblyFileVersionAttribute>
        <GenerateAssemblyInformationalVersionAttribute>false</GenerateAssemblyInformationalVersionAttribute>
      </PropertyGroup>
    
      <ItemGroup>
        <EmbeddedResource Include="Views\**" Exclude="bin\**;obj\**;**\*.xproj;packages\**;@(EmbeddedResource)" />
      </ItemGroup>
    
      <ItemGroup>
        <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="1.1.1" />
        <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.1.2" />
        <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="1.1.1" />
        <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="1.1.1" />
        <PackageReference Include="Microsoft.AspNetCore.WebUtilities" Version="1.1.1" />
        <PackageReference Include="Microsoft.AspNetCore.StaticFiles" Version="1.1.1" />
        <PackageReference Include="Microsoft.Extensions.FileProviders.Embedded" Version="1.1.0" />
      </ItemGroup>
    
    </Project>

I tried duplicating the `web.config` from the web application to the root of the class library but that didn't help. I've also tried moving over various entries from the web application's csproj file, but to no avail.

The entire project still builds, runs and publishes perfectly. I only see errors if I have a `cshtml` file from the library open in VS.

**Comment by pranavkm** [comment]
@NTaylorMullen isn't this the same issue @sebastienros is running in to?

**Comment by sebastienros** [comment]
I would say definitely.

@svallis try to use `<Project Sdk="Microsoft.NET.Sdk.Web">` in the meantime. It's kind of wrong but it should give you intellisense. Haven't tried it myself so far. Right @pranavkm?

**Comment by svallis** [comment]
@sebastienros Can confirm that brings back intellisense, thanks. Any downsides to switching the SDK? If there's anything I can help with re diagnosis/resolution please let me know.

**Comment by pranavkm** [comment]
@vijayrkn would probably know best - are there any side effects to changing the SDK of a class library to `Microsoft.NET.Sdk.Web`?

**Comment by vijayrkn** [comment]
There shouldn't be any side effects to changing the Net Sdk to Web Sdk. I don't know why intellisense is not working otherwise.

**Comment by svallis** [comment]
@vijayrkn Publishing inside a docker container breaks with the `Microsoft.NET.Sdk.Web` SDK specified. Changing back to `Microsoft.NET.Sdk` resolves the issue.

```
CSC : error CS5001: Program does not contain a static 'Main' method suitable for an entry point [/tmp/src/Library/Library.csproj]
```

**Comment by vijayrkn** [comment]
@NTaylorMullen - is websdk needed for intellisense ?

**Comment by pranavkm** [comment]
@svallis, sorry I missed seeing your comment about the build failure. Try setting `<OutputType>Library</OutputType>` to your project. The Web SDK defaults to making the project an executable.

**Comment by svallis** [comment]
Perfect, thanks @pranavkm!

**Comment by purplepiranha** [comment]
Excellent, finally working after trying various things all day.

Found that you also needed to reopen the solution for the intellisense to start working.

Thanks @pranavkm

**Comment by Eilon** [comment]
@rynowak - parking on your plate because this seems to be in the ballpark of what you've been up to.

**Comment by rynowak** [comment]
@Eilon sorry what's this about? Is this about tag helper intellisense in class libraries?

**Comment by Eilon** [comment]
@rynowak - that's what it looks like...?

**Comment by svallis** [comment]
Not just tag helper intellisense - all intellisence is lost when the SDK is set to `Microsoft.NET.Sdk` instead of `Microsoft.NET.Sdk.Web`. That may well be how things should be working, but the VS2017 auto-conversion tool was setting libraries to the former SDK, and therefore breaking intellisense. If the only problem that's really here is with the project converter though, then it may go away of its own accord.

**Comment by thiennn** [comment]
Look like the same issue https://github.com/aspnet/Tooling/issues/1013

**Comment by rynowak** [comment]
@NTaylorMullen fixed for preview2 right?

**Comment by NTaylorMullen** [comment]
Yup!

**Comment by rynowak** [comment]
Ok, thanks everyone for reporting this issue. This will be fixed in VS 2017 15.3-preview 2.

If you have further issues with tooling for Razor please report them via VS Feedback or on aspnet/Razor
